packages/gatsby-transformer-sharp/src/customize-schema.js

```diff
@@ -375,6 +380,182 @@ const fluidNodeType = ({
+      name: `ImageSharpGatsbyImage`,
+      fields: {
+        imageData: {
+          type: new GraphQLNonNull(GraphQLJSON),
+        },
+      },
+    }),
+    args: {
+      layout: {
+        type: ImageLayoutType,
+        defaultValue: `fixed`,
+        description: stripIndent`
+        The layout for the image.
+        FIXED: A static image sized, that does not resize according to the screen width
+        FLUID: The image resizes to fit its container. Pass a "sizes" option if it isn't going to be the full width of the screen. 
+        CONSTRAINED: Resizes to fit its container, up to a maximum width, at which point it will remain fixed in size.
+        `,
+      },
+      maxWidth: {
+        type: GraphQLInt,
+        description: stripIndent`
+        Maximum display width of generated files. 
+        The actual largest image resolution will be this value multipled by the largest value in outputPixelDensities
+        This only applies when layout = FLUID or CONSTRAINED. For other layout types, use "width"`,
+      },
+      maxHeight: {
+        type: GraphQLInt,
+      },
+      width: {
+        type: GraphQLInt,
+        description: stripIndent`
+        The display width of the generated image. 
+        The actual largest image resolution will be this value multipled by the largest value in outputPixelDensities
+        Ignored if layout = FLUID or CONSTRAINED, where you should use "maxWidth" instead.
+        `,
+      },
+      height: {
+        type: GraphQLInt,
+      },
+      placeholder: {
+        type: ImagePlaceholderType,
+        defaultValue: `blurred`,
+        description: stripIndent`
+        Format of generated placeholder image. 
+        DOMINANT_COLOR: a solid color, calculated from the dominant color of the image. 
+        BASE64: a blurred, low resolution image, encoded as a base64 data URI
+        TRACED_SVG: a low-resolution traced SVG of the image.
+        NONE: no placeholder. Set "background" to use a fixed background color.`,
+      },
+      tracedSVGOptions: {
+        type: PotraceType,
+        defaultValue: false,
+        description: `Options for traced placeholder SVGs. You also should set placeholder to SVG.`,
+      },
+      webP: {
+        type: GraphQLBoolean,
+        defaultValue: true,
+        description: `Generate images in WebP format as well as matching the input format. This is the default (and strongly recommended), but will add to processing time.`,
+      },
+      outputPixelDensities: {
+        type: GraphQLList(GraphQLInt),
+        description: stripIndent`
+        A list of image pixel densities to generate, for high-resolution (retina) screens. It will never generate images larger than the source, and will always a 1x image. 
+        Default is [ 1, 2 ] for fixed images, meaning 1x, 2x, 3x, and [0.25, 0.5, 1, 2] for fluid. In this case, an image with a fluid layout and width = 400 would generate images at 100, 200, 400 and 800px wide`,
+      },
```

Comment:
I'm a bit confused on what this would do for fluid? Why would I want to use it for fluid?

Reply:
You might want to produce lots of resolutions or hardly any. You mgiht decide you do want to support 3x. One option is if maxWidth actually means max resolution, rather than as it is now where it's multipled by the chosen densities

Reply:
For fluid this isn't working, this needs to be GraphQLFloat